### ISSUE TITLE
Does not appear to verify ECDSA (algo 13, 14) signatures

### ISSUE THREAD

**Issue by jpmens** [dns-inspector/dns-inspector#6]
Version 1.2.1 (build 11)

Query for `SOA` of zones which are signed with one or more ECDSA keys, e.g.

```
;; ANSWER SECTION:
isc.org.		2183	IN	DNSKEY	257 3 13 [key id = 7250]
isc.org.		2183	IN	DNSKEY	256 3 13 [key id = 27566]
isc.org.		2183	IN	RRSIG	DNSKEY 13 2 3600 20240914190228 20240831185841 7250 isc.org. [omitted]
```

If I query for SOA in DNS inspector, scroll down to _View DNSSEC Information_, both signature and chain are marked as `UNVERIFIED` respectively `UNTRUSTED`.

![rabbit-11307](https://github.com/user-attachments/assets/e9287841-d9a9-4ed9-ad36-1bf11c1d51de)

This appears to work with zones signed with RSA keys (e.g. algo 8)

**Comment by jpmens** [comment]
When querying for a `TXT` RR in the same zone using the same server, the response is validated. 

![image](https://github.com/user-attachments/assets/c212b614-ea0d-4c65-9e52-868be1134b40)

`NS` queries fail, `DS` and `DNSKEY` succeed.  🤷‍♂️

**Comment by jpmens** [comment]
Bug still in 1.2.2

**Comment by ecnepsnai** [comment]
Seems to only be for some zones. dns-inspector.com is signed with an ECDSA-P256 (algo 13) key and it verifies OK.

**Comment by ecnepsnai** [comment]
Root cause is that we're not using the canonical record data when building data to be signed.

Proper signed data:

```
00000000: 0006 0d02 0000 0708 6728 0107 6725 41e7  ........g(..g%A.
00000010: 86c9 0969 616e 7370 656e 6365 0363 6f6d  ...ianspence.com
00000020: 0009 6961 6e73 7065 6e63 6503 636f 6d00  ..ianspence.com.
00000030: 0006 0001 0000 0708 0040 0466 7265 6402  .........@.fred.
00000040: 6e73 0a63 6c6f 7564 666c 6172 6503 636f  ns.cloudflare.co
00000050: 6d00 0364 6e73 0a63 6c6f 7564 666c 6172  m..dns.cloudflar
00000060: 6503 636f 6d00 8c61 69cd 0000 2710 0000  e.com..ai...'...
00000070: 0960 0009 3a80 0000 0708                 .`..:.....
```

From DNSKit:

```
00000000: 0006 0d02 0000 0708 6728 0107 6725 41e7  ........g(..g%A.
00000010: 86c9 0969 616e 7370 656e 6365 0363 6f6d  ...ianspence.com
00000020: 0009 6961 6e73 7065 6e63 6503 636f 6d00  ..ianspence.com.
00000030: 0006 0001 0000 0708 002f 0466 7265 6402  ........./.fred.
00000040: 6e73 0a63 6c6f 7564 666c 6172 65c0 1603  ns.cloudflare...
00000050: 646e 73c0 338c 6169 cd00 0027 1000 0009  dns.3.ai...'....
00000060: 6000 093a 8000 0007 08                   `..:.....
```

Note the shorter response due to the data not being decompressed.

**Comment by ecnepsnai** [comment]
Fixed in 1.2.3
